### PR TITLE
feat(components): [button] add tag attribute

### LINKED ISSUE #8293: [Feature Request] [Enhancement] [button] A link looking like a button
Feature type: **`Enhancement`**

## Details

- Enhanced compone nt name: `button`

## Description

How can I place `<a href="…">` element looking like `<el-button>`?

`<el-button>` component always renders a `<button>` tag, but I want to render it as an `<a>` tag

### Additional comments

In other component libraries and frameworks there is a `tag` prop on "button" component, like this:

- Naive UI `<n-button>` has `tag` prop:
  https://www.naiveui.com/en-US/os-theme/components/button#tag.vue
- Buefy `<b-button>` has `tag` prop: 
  https://buefy.org/documentation/button/#tags
- Quasar `<q-btn>` has `type` prop. It can be one of `a/submit/button/reset`, or even `image/png`
  https://quasar.dev/vue-components/button#qbtn-api

### PR DESCRIPTION
**Pull request by tolking**
close #8293

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Review by chenxch — APPROVED** [review]

File: packages/components/button/src/button.ts [hunk 1]
```diff
@@ -104,6 +104,13 @@ export const buttonProps = buildProps({
     type: Boolean,
     default: undefined,
   },
+  /**
+   * @deprecated custom element tag
```
Comment (ryuhangyeong) [inline packages/components/button/src/button.ts:108]: ## Suggest

`@deprecated` >> `@description`

Comment (jw-foss) [reply packages/components/button/src/button.ts]: Nice catch!

**Comment by jw-foss** [comment]
@tolking can we give this a badge which tells the users that this will only be available above `2.3.0`?

**Review by jw-foss — APPROVED** [review]

**Review by ryuhangyeong — APPROVED** [review]

**Comment by tolking** [comment]
done. It is better to merge #11590 before this PR merger. Can you take a moment to look at it?

**Comment by btea** [comment]
@tolking The code seems to be conflicting.

**Review by btea — APPROVED** [review]

**Comment by dmix** [comment]
@tolking thanks for doing that 😄 

This was an annoying issue for us for testing using Capybara in our Rails app. We can now get rid of our git fork / custom implementation.

**Pull request merged**